pkg/collector/containerdlogs_collector.go

```diff
@@ -0,0 +1,106 @@
+package collector
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// ContainerDLogsCollector defines a ContainerDLogs Collector struct
+type ContainerDLogsCollector struct {
+	BaseCollector
+}
+
+// ContainersList defines configuration for list of containers and their details
+type ContainersList struct {
+	ContainersList []ContainerInformation `json:"containers"`
+}
+
+// ContainerInformation defines configuration of information available for each container
+type ContainerInformation struct {
+	ID     string            `json:"id"`
+	Labels map[string]string `json:"labels"`
+}
+
+var _ interfaces.Collector = &ContainerDLogsCollector{}
+
+// NewContainerDLogsCollector is a constructor
+func NewContainerDLogsCollector(exporter interfaces.Exporter) *ContainerDLogsCollector {
+	return &ContainerDLogsCollector{
+		BaseCollector: BaseCollector{
+			collectorType: ContainerDLogs,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *ContainerDLogsCollector) Collect() error {
+	containerLogs := strings.Fields(os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST"))
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	output, err := utils.RunCommandOnHost("crictl", "ps", "-a", "-o", "json")
+	if err != nil {
+		return err
+	}
+
+	var containers ContainersList
+	if err = json.Unmarshal([]byte(output), &containers); err != nil {
+		log.Fatalf("unable to read containers json file: %v", err)
+	}
+
+	for _, containerLog := range containerLogs {
+		containerLogParts := strings.Split(containerLog, "/")
+		targetNameSpace := containerLogParts[0]
+		containerNames := map[string]string{}
+
+		for _, container := range containers.ContainersList {
+			containerID := container.ID
+			containerNameSpace, ok := container.Labels["io.kubernetes.pod.namespace"]
+			if !ok {
+				log.Printf("Unable to retrieve namespace of container: %+v", containerID)
+				continue
+			}
+
+			containerName, ok := container.Labels["io.kubernetes.container.name"]
+			if !ok {
+				log.Printf("Unable to retrieve name of container: %+v", containerID)
+				continue
+			}
+
+			if len(containerLogParts) == 2 {
+				if strings.HasPrefix(containerName, containerLogParts[1]) && containerNameSpace == targetNameSpace {
+					containerNames[containerID] = containerName
+				}
+			} else {

```

Comment:
💡 This is interesting, what is the this `if-else` clause doing? 

i.e. in both cases it achieves `containerNames[containerID] = containerName` resultset?

```
		if len(containerLogParts) == 2 {
			if strings.HasPrefix(containerName, containerLogParts[1]) && containerNameSpace == targetNameSpace {
				containerNames[containerID] = containerName
			}
		} else {
			if containerNameSpace == targetNameSpace {
				containerNames[containerID] = containerName
			}
		}
```

Reply:
Based on the README file in the root, users can specify containers like this:
```
az aks kollect \
-g MyResourceGroup \
-n MyManagedCluster \
--container-logs "mynamespace1/mypod1 myns2"
```
So `--container-logs` flag accepts two formats: "namespace/container" or just "namespace".
Here is logic is, if the length of `containerLogParts` is 2, meaning the input format matches "namespace/container", we need to make sure both namespace and container name match. Otherwise, we only need to check if the namespace match.

Reply:
BTW, if you check the existing container collector, it's the same logic over there.

Reply:
Isn't it a duplicate of `DIAGNOSTIC_CONTAINERLOGS_LIST`? We might want to remove this

Reply:
--container-logs "mynamespace1/mypod1 myns2" is used with az aks kollect (the az cli command), under the hood this just sets the value of DIAGNOSTIC_CONTAINERLOGS_LIST before deploying